docs/modules/ROOT/pages/usage/daemon.adoc

```diff
@@ -0,0 +1,84 @@
+= Daemon
+
+You can reduce the RuboCop boot time by using the `--daemon` option.
+
+The daemon option speeds up the launch of the `rubocop` command by daemonizing
+the process that loaded the RuboCop runtime production files (i.e. `require 'rubocop'`).
+
+NOTE: The feature cannot be used on JRuby and Windows that do not support fork.
+
+== Run with Daemon
+
+There are two ways to enable daemon.
+
+- `rubocop --daemon` ... If daemon process has not started yet,
+start daemon process and execute inspection with daemon.
+- `rubocop --start-daemon` ... Just start daemon process.
+
+When daemon is started, it outputs the host and port.
+
+```console
+% rubocop --start-daemon
+Daemon process listen on 127.0.0.1:55772.
+```
+
+NOTE: The `rubocop` command is executed using daemon server when daemon process is started.
+Whenever daemon process is not running, it will load the RuboCop runtime files and execute.
+(same behavior as 1.30 and below)
+
+If daemon is already running, just display the PID. The new daemon will not start.
+
+```console
+% rubocop --start-daemon
+Daemon process (16060) is already running.
+```
+
+A started daemon process name is `rubocop --daemon` with project directory path.
+
+```console
+% ps aux | grep 'rubocop --daemon'
+user             16060   0.0  0.0  5078568   2264   ??  S     7:54AM   0:00.00 rubocop --daemon /Users/user/src/github.com/rubocop/rubocop
+user             16337   0.0  0.0  5331560   2396   ??  S    23:51PM   0:00.00 rubocop --daemon /Users/user/src/github.com/rubocop/rubocop-rails
+```
+
+== Restart Daemon
+
+The started daemon does not reload the configuration file.
+You will need to restart daemon when you upgrade RuboCop or change
+the RuboCop configuration.
+
+```console
+% rubocop --restart-daemon
+Daemon process listen on 127.0.0.1:55822.
+```
+
+This may be supported so that no reboot is required in future.
+
+== Command Line Options
+
+These are options for daemon operations.
+
+* `--daemon` ... If daemon process has not started yet, start the
+daemon process and execute inspection with daemon. You can specify
+the daemon host and port with the $RUBOCOP_DAEMON_HOST and
+the $RUBOCOP_DAEMON_PORT environment variables.
+* `--no-daemon` ... If daemon process has been started, stop the
+daemon process and execute inspection with daemon.
+* `--restart-daemon` ... Restart daemon process.
+* `--start-daemon` ... Start daemon process.
+* `--stop-daemon` ... Stop daemon process.
+* `--daemon-status` ... Show daemon status.
+
+== Environment Variables
+
+You can change the startup host and port of daemon process with
+environment variables.
+
+* `$RUBOCOP_SERVER_HOST`
+* `$RUBOCOP_SERVER_PORT`
+
+The following is an example:
+
+```console
+% RUBOCOP_SERVER_PORT=98989 rubocop --start-daemon
```

Comment:
Generally it's not a bad idea to be able to specify the host and port directly via command-line flags as well. That's part of the reason why I like a separate command for managing the server - otherwise we end with a ton of extra options in `rubocop` itself.